records_mover/db/vertica/unloader.py

```diff
@@ -97,7 +103,11 @@ def unload_to_s3_directory(self,
         processing_instructions = unload_plan.processing_instructions
         try:
             s3_sql = self.aws_creds_sql(aws_creds.access_key, aws_creds.secret_key)
-            self.db.execute(s3_sql)
+            if not self.db_conn.in_transaction():
+                with self.db_conn.begin():
+                    self.db_conn.execute(text(s3_sql))
+            else:
+                self.db_conn.execute(text(s3_sql))
```

Comment:
yeah this kind of confusion is the downside of the approach we took. where you have to check the state of the passed in connection rather than explicitly always managing it yourself.  But once we move past this interim of still trying to support the old ways it should get better i think